src/lib/snack-bar/snack-bar.ts

```diff
@@ -88,10 +97,9 @@ export class MdSnackBar {
                                     overlayRef: OverlayRef): MdSnackBarRef<T> {
     let portal = new ComponentPortal(component);
     let contentRef = container.attachComponentPortal(portal);
-    let snackBarRef = <MdSnackBarRef<T>> new MdSnackBarRef(contentRef.instance, overlayRef);
-
-    this._snackBarRef = snackBarRef;
-    return snackBarRef;
+    return <MdSnackBarRef<T>> new MdSnackBarRef(contentRef.instance,
```

Comment:
I want to say that you shouldn't need the typecast, but I'm too jet lagged to think any deeper about it.